Declarative encoding API from QML

oKcerG opened the pull request:
This adds a better api for displaying images in QML.
To display an encoded qrcode from QML, we now can do:  
`Image { source: "image://QZXing/qrcode/<data>" }`

This allows different encoded images to be displayed simultaneously.

The `qrcode` part of the url is there to allow potential other future formats.
We could also maybe implement encoding options by doing for example `qrcode?errorCorrectionLevel=low` (this is not implemented in this PR).

The `image://QZXing/latestEncoded` url no longer works, but since the encoding is not working yet, I don't think people relied on that.

Support for requested size is still missing in the image provider. I changed the image provider to return a `QImage` instead of a `QPixmap`.

I changed `QZXing::encodeData` to be static, it didn't need to be a member function.

Closes #31

ftylitak commented:
@oKcerG thank you very much for your contribution!

I will look at your changes in the coming days and merge it as soon as possible.

oKcerG commented:
@ftylitak any comment on this PR?

ftylitak commented:
Sorry @oKcerG for the delay,

i had no time to thoroughly look at it so i have postponed it. Any way, i was not sure regarding the automatic barcode generation as soon as the text changes, though it is funny so i will accept it. Apart from that it is ok :)

Thank you for your support!

oKcerG commented:
Yeah I changed the demo this way cause I thought it was a bit more impressive and showcased the power of QML :) It also simplified the UI code.
Obviously this doesn't have much to do with this new feature. Users can still generate a qr code on demand (with a button, after a delay, etc.) and not automatically.
